### PR TITLE
Summary list component

### PR DESCRIPTION
**Pull request by jake-bassett**
## Description
This adds a summary list component that can be used to list out labelled data.

A few other miscellaneous things (bad, Jake!):
* Added two icons for internal and external
* Fixed bands to draw behind series on cartesians
* Re-ordered components public-api

### CODE REVIEW

**Comment by anandtiwary** [comment]
Could you please share the screenshot?

File: projects/components/src/public-api.ts [hunk 1]
```diff
@@ -2,6 +2,10 @@
  * Public API Surface of components
  */
```
Comment (aaron-steinfeld) [inline projects/components/src/public-api.ts:4]: 👀

File: projects/components/src/summary-list/summary-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { IconType } from '@hypertrace/assets-library';
+import { PrimitiveValue } from '@hypertrace/common';
+import { startCase } from 'lodash-es';
+import { SummaryItem } from './summary-list-api';
+
+@Component({
+  selector: 'ht-summary-list',
```
Comment (aaron-steinfeld) [inline projects/components/src/summary-list/summary-list.component.ts:8]: I don't love the name (when do I ever?). We already have a list-view, card-list, summary-values, table - so from the names, it's not clear how this one is different and when to use which.

File: projects/components/src/summary-list/summary-list.component.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  public title?: string;
+
+  @Input()
+  public icon?: IconType;
+
+  @Input()
+  public items?: SummaryItem[] = [];
+
+  public getFormattedLabel(item: SummaryItem): string {
+    return startCase(item.label.trim().replace('-', ' ').replace('_', ' ').toLowerCase());
```
Comment (aaron-steinfeld) [inline projects/components/src/summary-list/summary-list.component.ts:38]: I thought start case already handled most, if not all of these - could you check?

File: projects/observability/src/shared/components/cartesian/d3/chart/cartesian-chart.ts [hunk 4]
```diff
@@ -439,11 +439,11 @@ export class DefaultCartesianChart<TData> implements CartesianChart<TData> {
     ];
 
     this.allCartesianData = [
-      ...this.allSeriesData,
       ...this.bands.map(
         band =>
           new CartesianBand(this.d3Utils, this.domRenderer, band, this.scaleBuilder, this.getTooltipTrackingStrategy())
-      )
+      ),
+      ...this.allSeriesData
```
Comment (aaron-steinfeld) [inline projects/observability/src/shared/components/cartesian/d3/chart/cartesian-chart.ts:446]: Seems unrelated? intentional?

File: projects/components/src/summary-list/summary-list.component.ts (hunk 2 above)
Comment (jake-bassett) [reply projects/components/src/summary-list/summary-list.component.ts:8]: Yeah, I was scanning through our components and thinking the same thing. Open to suggestions. :-)

File: projects/observability/src/shared/components/cartesian/d3/chart/cartesian-chart.ts (hunk 4 above)
Comment (jake-bassett) [reply projects/observability/src/shared/components/cartesian/d3/chart/cartesian-chart.ts:446]: This is unrelated (sorry, I suck), but intentional. As I was working on this feature I was on a page that showed bands and I wanted to fix the front to back ordering since it was bothering me.

File: projects/components/src/summary-list/summary-list.component.ts [hunk 5]
```diff
@@ -35,7 +35,7 @@ export class SummaryListComponent {
   public items?: SummaryItem[] = [];
 
   public getFormattedLabel(item: SummaryItem): string {
-    return startCase(item.label.trim().replace('-', ' ').replace('_', ' ').toLowerCase());
+    return startCase(item.label.trim().toLowerCase());
```
Comment (aaron-steinfeld) [inline projects/components/src/summary-list/summary-list.component.ts:38]: trim is definitely still taken care of by startCase, the most intelligent of all cases. lower case may not be though.

File: projects/components/src/summary-list/summary-list.component.ts (hunk 2 above)
Comment (aaron-steinfeld) [reply projects/components/src/summary-list/summary-list.component.ts:8]: Ah, of course - my own words turned against me. In that case, carry on.

**Review by aaron-steinfeld — COMMENTED** [review]
Please add a test for the new component, hopefully someone can come up with a more inspired name in the meantime and if not we'll go with this.

File: projects/components/src/summary-list/summary-list.component.test.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { SummaryListComponent } from '@hypertrace/components';
+import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
+
+describe('Summary List component', () => {
+  let spectator: Spectator<SummaryListComponent>;
+
+  const createComponent = createComponentFactory({
+    component: SummaryListComponent,
```
Comment (aaron-steinfeld) [inline projects/components/src/summary-list/summary-list.component.test.ts:8]: For a component with inputs and ouputs, using a hostfactory allows you to provide inputs and then verify it rendered correctly. This test only verifies the logic directly rather than for a given set of inputs it renders the html we want - that's what we're generally looking for in a component's tests - the functions are implementation details.

**Review by aaron-steinfeld — APPROVED** [review]

File: projects/components/src/summary-list/summary-list.component.test.ts [hunk 7]
```diff
@@ -1,32 +1,79 @@
...
 
   test('gets value array', () => {
-    expect(spectator.component.getValuesArray(1)).toEqual([1]);
-    expect(spectator.component.getValuesArray([1])).toEqual([1]);
-    expect(spectator.component.getValuesArray('two')).toEqual(['two']);
-    expect(spectator.component.getValuesArray(['two', 'three'])).toEqual(['two', 'three']);
-    expect(spectator.component.getValuesArray(true)).toEqual([true]);
-    expect(spectator.component.getValuesArray([true, false])).toEqual([true, false]);
+    const values = spectator.queryAll('li').map(e => e.textContent);
+    expect(values).toEqual(['0', '0', '1', '2', 'zero', 'zero', 'one', 'two', 'three', 'true', 'true', 'false']);
```
Comment (aaron-steinfeld) [inline projects/components/src/summary-list/summary-list.component.test.ts:77]: https://www.youtube.com/watch?v=2IPAOxrH7Ro&t=158s

**Review by aaron-steinfeld — APPROVED** [review]

File: projects/components/src/summary-list/summary-list.component.test.ts (hunk 7 above)
Comment (jake-bassett) [reply projects/components/src/summary-list/summary-list.component.test.ts:77]: It was worth updating this test just to see that video.
